### PR TITLE
fix: Remove circular dependency

### PR DESCRIPTION
**Pull request by yaraslau-kavaliou**
Ticket - https://oat-sa.atlassian.net/browse/REL-1089

### CODE REVIEW

File: actions/TestTaker.php [hunk 1]
```diff
@@ -163,6 +163,7 @@ public function editSubject()
         $myForm = $myFormContainer->getForm();
 
         $subjectUri = $subject->getUri();
+        $userService = tao_models_classes_UserService::singleton();
```
Comment (augustas) [inline actions/TestTaker.php:166]: Use of `singleton()` is deprecated, this service can be pulled from ServiceLocator. Easy to make it better than before: 
```
$this->getServiceLocator()->get(tao_models_classes_UserService::SERVICE_ID)
```

**Review by augustas — CHANGES_REQUESTED** [review]
Since the dependency on https://github.com/oat-sa/extension-tao-group is present, please add the requirement in [composer.json](https://github.com/oat-sa/extension-tao-testtaker/blob/master/composer.json#L58)

File: actions/TestTaker.php (hunk 1 above)
Comment (yaraslau-kavaliou) [reply actions/TestTaker.php]: done

**Review by yaraslau-kavaliou — COMMENTED** [review]
@augustas If we add dependency to `extension-tao-group` we will end up with circular dependency because `extension-tao-group` already has dependency from `extension-tao-testtaker` - https://github.com/oat-sa/extension-tao-group/blob/master/composer.json  :)

File: actions/TestTaker.php [hunk 2]
```diff
@@ -226,10 +226,11 @@ public function editSubject()
             $this->setData('reload', true);
         }
 
-        $assignmentService = $this->getServiceLocator()->get(AssignmentService::SERVICE_ID);
+        $currentUser = $userService->getCurrentUser();
+        $groupManagerRole = new core_kernel_classes_Resource(self::GROUP_MANAGER_ROLE);
         if (
             common_ext_ExtensionsManager::singleton()->isEnabled('taoGroups') &&
-            get_class($assignmentService) == GroupAssignment::class
+            $userService->userHasRoles($currentUser, $groupManagerRole)
```
Comment (gabrielfs7) [inline actions/TestTaker.php:233]: Before we did not have role restriction for this. Can we make sure that expected roles by business will still be able to manage this? As you know, when you create a user the role `Group Manager` is not available for selection. 

So I would make sure that business provide a list of roles they assume that should still have access for this and make sure they have `http://www.tao.lu/Ontologies/TAOGroup.rdf#GroupsManagerRole` linked to them (if necessary).

**Pull request closed without merging**